### PR TITLE
Get Icon SVGs form Figma api

### PR DESCRIPTION
**Pull request by SiAdcock**
## What is the purpose of this change?

This change introduces a new script which queries the Figma API in order to get the SVG source of the icons provided by Source. This is the first part of a piece of work to automate the synchronisation of icons between Figma and the react component library.

## What does this change?

-   Add a new script to get icons from the Figma API and write to disk
-   Add the 24x24 icons to the repo

### CODE REVIEW

File: scripts/get-icons/index.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import axios from 'axios';
+import { existsSync, mkdirSync, writeFileSync } from 'fs';
+
+const ICON_FILE = 'Ai7AELHC6KCz38qKZkvuHo';
+
+const ICON_FRAME = 'UI icons 24(w)x24(w)';
+
+const OUTPUT_DIR = './svgs';
```
Comment (SiAdcock) [inline scripts/get-icons/index.ts:8]:
```suggestion
const OUTPUT_DIR = '../../packages/@guardian/src-icons/svgs';
```

File: scripts/get-icons/index.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	name: string;
+	node_id: string;
+}
+
+interface NodeWithUrl extends Node {
+	url: string;
+}
+
+const getUrlsForNodes = (nodes: Node[]): Promise<NodeWithUrl[]> => {
+	return new Promise((resolve) => {
```
Comment (SiAdcock) [inline scripts/get-icons/index.ts:38]: Just to help me understand, why do we need to wrap the `axios` response in a Promise? My brain is thinking it's already returning a `Promise` 🤔  😄

File: scripts/get-icons/index.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	});
+};
+
+const getAndWriteSVGForNode = (node: NodeWithUrl) => {
+	return new Promise((resolve) => {
+		return axios
+			.get(node.url)
+			.then((res) => {
+				return resolve(
+					writeFileSync(`svgs/${node.name}.svg`, res.data),
```
Comment (SiAdcock) [inline scripts/get-icons/index.ts:63]:
```suggestion
					writeFileSync(`OUTPUT_DIR/${node.name}.svg`, res.data),
```

**Comment by SiAdcock** [comment]
This is looking pretty awesome, @jamie-lynch! 🥇 

Shall we keep a list of what else we think this needs so we can pick up the last little bits later?

- [ ] Remove `height`, `width` and `fill` attributes from `<svg>`s
- [ ] Pass through [SVGOMG](https://jakearchibald.github.io/svgomg/)
- [ ] Add README
- [ ] Generate React component modules from `.svg` files

Anything else?

File: scripts/get-icons/package.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "get-icons",
+  "version": "1.0.0",
+  "private": true,
+  "license": "MIT",
```
Comment (SiAdcock) [inline scripts/get-icons/package.json:5]:
```suggestion
  "license": "Apache-2.0",
```

File: scripts/get-icons/index.ts (hunk 2 above)
Comment (jamie-lynch) [reply scripts/get-icons/index.ts]: We don't 👍 Good spot. I've removed it now and it all still works as expected.

**Comment by jamie-lynch** [comment]
> Shall we keep a list of what else we think this needs so we can pick up the last little bits later?

Great shout. I've created a new card on the backlog to track this. 🎟️

File: scripts/get-icons/index.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
+import axios from 'axios';
+import { existsSync, mkdirSync, writeFileSync } from 'fs';
+
+const ICON_FILE = 'Ai7AELHC6KCz38qKZkvuHo';
+
+const ICON_FRAME = 'UI icons 24(w)x24(w)';
+
+const OUTPUT_DIR = '../../packages/@guardian/src-icons/svg';
```
Comment (SiAdcock) [inline scripts/get-icons/index.ts:8]: Sorry, on thinking about it, would it be better to pluralise this directory name? We tend to pluralise things elsewhere in this repo (`components`, `foundations`, `icons`, `'libs`)

```suggestion
const OUTPUT_DIR = '../../packages/@guardian/src-icons/svgs';
```

**Comment by SiAdcock** [comment]
I can't approve this PR because I created it! But feel free to self-approve as I think we're both very happy with it 😄 

👍 👍 👍 👍 👍 👍 👍

**Pull request merged**